packages/mucoll-stack/package.py

```diff
@@ -108,9 +112,12 @@ class MucollStack(BundlePackage, Key4hepPackage):
         depends_on('ninja')
         depends_on('doxygen')
         depends_on('gdb')
-        depends_on('llvm')
-        depends_on('onnx')
-        depends_on('xgboost')
+    
+    depends_on('llvm', when='+llvm')
+    depends_on('onnx', when='+ml')
```

Comment:
I can't comment directly below, so I am attaching it here. I am wondering whether `py-onnx` and `py-onnxruntime` should also go into the `+ml` variant rather than the `+pytools` one. Specifically also because `py-onnxruntime` is (slightly paradoxically) also the one that provides the c++ interface for OnnxRuntime.

Reply:
Agreed!